File: app/src/main/java/com/movienight/data/Movie.java
package com.movienight.data;

import com.google.gson.annotations.SerializedName;

import android.os.Parcel;
import android.os.Parcelable;

/**
 * @author anna
 */

public class Movie implements Parcelable {

    public void setMovieTitle(String movieTitle) {
        this.movieTitle = movieTitle;
    }

    public void setMovieRating(String movieRating) {
        this.movieRating = movieRating;
    }

    public void setMovieId(int movieId) {
        this.movieId = movieId;
    }

    public void setMoviePosterImagePath(String moviePosterImagePath) {
        this.moviePosterImagePath = moviePosterImagePath;
    }

    public void setMoviePlot(String moviePlot) {
        this.moviePlot = moviePlot;
    }

    public void setMovieRelease(String movieRelease) {
        this.movieRelease = movieRelease;
    }

    @SerializedName("title")
    private String movieTitle;

    @SerializedName("vote_average")
    private String movieRating;

    @SerializedName("id")
    private int movieId;

    @SerializedName("poster_path")
    private String moviePosterImagePath;

    @SerializedName("overview")
    private String moviePlot;

    @SerializedName("release_date")
    private String movieRelease;

    protected Movie(Parcel in) {
        movieTitle = in.readString();
        movieRating = in.readString();
        movieId = in.readInt();
        moviePosterImagePath = in.readString();
        moviePlot = in.readString();
        movieRelease = in.readString();
    }

    public Movie() {

    }

    public static final Creator<Movie> CREATOR = new Creator<Movie>() {
        @Override
        public Movie createFromParcel(Parcel in) {
            return new Movie(in);
        }

        @Override
        public Movie[] newArray(int size) {
            return new Movie[size];
        }
    };

    public String getMovieTitle() {
        return movieTitle;
    }

    public String getMovieRating() {
        return movieRating;
    }

    public String getMoviePosterImagePath() {
        return moviePosterImagePath;
    }

    public String getMoviePlot() {
        return moviePlot;
    }

    public String getMovieRelease() {
        return movieRelease;
    }

    public int getMovieId() {
        return movieId;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel parcel, int i) {
        parcel.writeString(movieTitle);
        parcel.writeString(movieRating);
        parcel.writeInt(movieId);
        parcel.writeString(moviePosterImagePath);
        parcel.writeString(moviePlot);
        parcel.writeString(movieRelease);
    }
}

File: app/src/main/java/com/movienight/data/Movie.kt
package com.movienight.data

import android.arch.persistence.room.Entity
import android.arch.persistence.room.PrimaryKey
import android.os.Parcel
import android.os.Parcelable

@Entity(tableName = "movie_table")
data class Movie(val title: String,
        val vote_average: String? = "0",
        @PrimaryKey val id: Int,
        val poster_path: String? = null,
        val overview: String? = null,
        val release_date: String? = null) : Parcelable {

    constructor(parcel: Parcel) : this(
            parcel.readString(),
            parcel.readString(),
            parcel.readInt(),
            parcel.readString(),
            parcel.readString(),
            parcel.readString()) {
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(title)
        parcel.writeString(vote_average)
        parcel.writeInt(id)
        parcel.writeString(poster_path)
        parcel.writeString(overview)
        parcel.writeString(release_date)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<Movie> {
        override fun createFromParcel(parcel: Parcel): Movie {
            return Movie(parcel)
        }

        override fun newArray(size: Int): Array<Movie?> {
            return arrayOfNulls(size)
        }
    }
}